vllm/v1/engine/async_llm.py

```diff
@@ -177,33 +179,45 @@ async def add_request(
     ) -> asyncio.Queue[RequestOutput]:
         """Add new request to the AsyncLLM."""
 
-        # 1) Create a new output queue for the request.
+        # Create a new output queue for the request.
         queue: asyncio.Queue[RequestOutput] = asyncio.Queue()
 
-        # 2) Fan out child requests (for n>1)
-        parent_req = ParentRequest.from_params(request_id, params)
+        # Convert Input --> Request.
+        request = self.processor.process_inputs(request_id, prompt, params,
+                                                arrival_time, lora_request,
+                                                trace_headers,
+                                                prompt_adapter_request,
+                                                priority)
+
         n = params.n if isinstance(params, SamplingParams) else 1
-        for idx in range(n):
-            if parent_req is not None:
+
+        if n == 1:
+            await self._add_request(request, None, 0, queue)
+
+        else:
```

Comment:
Not sure if worthwhile, but perhaps you could rewrite this section to make the `else` redundant:

```suggestion
        if n == 1:
            await self._add_request(request, None, 0, queue)
            return queue
        
        # Fan out child requests (for n>1).
        parent_req = ParentRequest(request_id, params)
        for idx in range(n):
            request_id, params = parent_req.get_child_info(idx)
            child_req = request if idx == n - 1 else copy(request)
            child_req.request_id = request_id
            child_req.sampling_params = params
            await self._add_request(child_req, parent_req, idx, queue)
                
        return queue
```

I like this because it expresses that there is a "quick exit" when `n==1` and a longer process otherwise.